notebooks/config.ipynb

```diff
@@ -964,12 +971,831 @@
+   "outputs": [],
+   "source": [
+    "%%bash\n",
+    "cp fixtures/config/base-config.nml tmp/config-copy.nml"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "02c75cf7-9f24-44a0-97c6-abd7b44fb60f",
+   "metadata": {},
+   "source": [
+    "When these two files are compared, `True` is returned and the log reports no differences.\n",
+    "<!--cell 62-->"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 32,
+   "id": "8b495735-2396-436e-87db-d886ac6769fa",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "[2024-10-11T10:36:12]     INFO - fixtures/config/base-config.nml\n",
+      "[2024-10-11T10:36:12]     INFO + tmp/config-copy.nml\n",
+      "[2024-10-11T10:36:12]     INFO ---------------------------------------------------------------------\n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "True"
+      ]
+     },
+     "execution_count": 32,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "config.compare(\n",
+    "    config_1_path='fixtures/config/base-config.nml',\n",
+    "    config_2_path='tmp/config-copy.nml',\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e14c0158-baa4-45d0-a86c-d9a245021229",
+   "metadata": {},
+   "source": [
+    "## Validating Configs\n",
+    "\n",
+    "The `config.validate()` function checks if a given config conforms to a specified JSON schema.\n",
+    "<!--cell 64-->"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 33,
+   "id": "2eb5b2ce-bebd-449d-90a5-764484aa03aa",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Help on function validate in module uwtools.api.config:\n",
+      "\n",
+      "validate(schema_file: Union[pathlib.Path, str], config: Union[dict, str, uwtools.config.formats.yaml.YAMLConfig, pathlib.Path, NoneType] = None, stdin_ok: bool = False) -> bool\n",
+      "    Check whether the specified config conforms to the specified JSON Schema spec.\n",
+      "\n",
+      "    If no config is specified, ``stdin`` is read and will be parsed as YAML and then validated. A\n",
+      "    ``dict`` or a YAMLConfig instance may also be provided for validation.\n",
+      "\n",
+      "    :param schema_file: The JSON Schema file to use for validation.\n",
+      "    :param config: The config to validate.\n",
+      "    :param stdin_ok: OK to read from ``stdin``?\n",
+      "    :return: ``True`` if the YAML file conforms to the schema, ``False`` otherwise.\n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "help(config.validate)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1eeca5ad-e969-4670-bddd-8e20b38f3d6c",
+   "metadata": {},
+   "source": [
+    "Below is an example of a schema used to validate a config. It ensures that the required keys are present and the value types match expectations.\n",
```

Comment:
It would be great if we could link to JSON Schema docs in this text block.

Reply:
Agreed, I added the link to this block.